From c78fed7729d9e80ea4af4f8c739113e5cf2e1f11 Mon Sep 17 00:00:00 2001
From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Date: Fri, 13 May 2016 11:08:17 +0800
Subject: [PATCH 11/18] Ignore rild modem devices.

---
 src/nm-manager.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/nm-manager.c b/src/nm-manager.c
index ada30d9fc6..dce83b50e3 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -3162,6 +3162,14 @@ add:
 		gboolean ignore = FALSE;
 		gs_free_error GError *error = NULL;
 
+		/* Ignore rild modem devices, which will be handled by their modem parent */
+		if (g_strstr_len (plink->name, NM_STRLEN ("rmnet"), "rmnet") ||
+			g_strstr_len (plink->name, NM_STRLEN ("rev_rmnet"), "rev_rmnet") ||
+			g_strstr_len (plink->name, NM_STRLEN ("ccmni"), "ccmni")) {
+			_LOGW (LOGD_PLATFORM, "Ignoring rild modem device: %s", plink->name);
+			return;
+		}
+
 		device = nm_device_factory_create_device (factory, plink->name, plink, NULL, &ignore, &error);
 		if (!device) {
 			if (!ignore) {
-- 
2.20.1

